plugins/performance-lab/includes/site-health/audit-enqueued-assets/hooks.php

```diff
@@ -13,107 +13,109 @@
+			if ( is_string( $type ) && '' !== $type && 'module' === strtolower( $type ) ) {
+				continue;
+			}
 
-/**
- * Audit enqueued and printed styles in the frontend. Ignore /wp-includes styles.
- *
- * It will save information in a transient for 12 hours.
- *
- * @since 1.0.0
- *
- * @global WP_Styles $wp_styles The WP_Styles current instance.
- */
-function perflab_aea_audit_enqueued_styles(): void {
-	if ( ! is_admin() && is_front_page() && current_user_can( 'view_site_health_checks' ) && false === get_transient( 'aea_enqueued_front_page_styles' ) ) {
-		global $wp_styles;
-		$enqueued_styles = array();
-		foreach ( $wp_styles->done as $handle ) {
-			$style = $wp_styles->registered[ $handle ];
-
-			if ( ! $style->src || false !== strpos( $style->src, 'wp-includes' ) ) {
+			// Skip external script with a "type" attribute that is not JavaScript.
+			if ( is_string( $type ) && '' !== $type && 'text/javascript' !== strtolower( $type ) ) {
 				continue;
 			}
 
-			// Check if we already have the style's path ( part of a refactor for block styles from 5.9 ).
-			if (
-				isset( $style->extra['path'] ) &&
-				is_string( $style->extra['path'] ) &&
-				'' !== $style->extra['path']
-			) {
-				$path = $style->extra['path'];
-			} else { // Fallback to getting the path from the style's src.
-				$path = perflab_aea_get_path_from_resource_url( $style->src );
-				if ( '' === $path ) {
-					continue;
-				}
+			$size = perflab_aea_get_asset_content_length( $src );
+			if ( false !== $size ) {
+				$assets['scripts'][] = array(
+					'src'  => $src,
+					'size' => $size,
+				);
+			}
+		} elseif ( 'LINK' === $tag ) {
+			$rel = $processor->get_attribute( 'rel' );
+			if ( ! is_string( $rel ) || 'stylesheet' !== strtolower( $rel ) ) {
+				continue;
 			}
 
-			// Add any extra data (inlined) that was passed with the style.
-			$inline_size = 0;
-			if (
-				isset( $style->extra['after'] ) &&
-				is_array( $style->extra['after'] )
-			) {
-				foreach ( $style->extra['after'] as $extra ) {
-					$inline_size += ( is_string( $extra ) ) ? mb_strlen( $extra, '8bit' ) : 0;
-				}
+			$href = $processor->get_attribute( 'href' );
+			if ( ! is_string( $href ) ) {
+				continue;
 			}
 
-			$enqueued_styles[] = array(
-				'src'  => $style->src,
-				'size' => wp_filesize( $path ) + $inline_size,
-			);
+			$size = perflab_aea_get_asset_content_length( $href );
+			if ( false !== $size ) {
+				$assets['styles'][] = array(
+					'src'  => $href,
+					'size' => $size,
+				);
+			}
 		}
-		set_transient( 'aea_enqueued_front_page_styles', $enqueued_styles, 12 * HOUR_IN_SECONDS );
+	}
+
+	if ( 0 !== count( $assets['scripts'] ) ) {
+		set_transient( 'aea_enqueued_front_page_scripts', $assets['scripts'], 12 * HOUR_IN_SECONDS );
+	}
+	if ( 0 !== count( $assets['styles'] ) ) {
+		set_transient( 'aea_enqueued_front_page_styles', $assets['styles'], 12 * HOUR_IN_SECONDS );
 	}
```

Comment:
Shouldn't the transients always be set? Otherwise, if they expire or are not set then this will result in the logic in `perflab_aea_audit_blocking_assets()` being called with every admin page load.

```suggestion
	set_transient( 'aea_enqueued_front_page_scripts', $assets['scripts'], 12 * HOUR_IN_SECONDS );
	set_transient( 'aea_enqueued_front_page_styles', $assets['styles'], 12 * HOUR_IN_SECONDS );
```